### PR TITLE
Search Fixes

### PR DESCRIPTION
**Pull request by ab7289**
This PR fixes some inaccuracies in the fetch_geocode and thus the search functionality.

- Differences between the typed in address and the address returned from google (apartment upload)
- Updated the `normalize_us_address` to not return addresses that were not in NY state
- adds a `locality` field on the location class that is used when the 'city' and 'locality' returned from google are not the same (i.e. Long Island City (city) in Queens(locality))
- updated the search so searching for 'Queens, NY' (a more general search), will also return results in Long Island City, which is part of Queens.
- Searches for specific neighborhoods though will only return results for that neighborhood (i.e. a search for Long Island City will not return results in Ridgewood or Flushing)
- cleaned up some of our code
- added test cases

Fixes test day bugs: #292 , #277 , #279, and #270

### CODE REVIEW

**Comment by ab7289** [comment]
This PR relies on changes made in #314 so please do not merge it until after that one has been merged into develop.

File: external/googleapi/tests.py [hunk 1]
```diff
@@ -569,3 +606,47 @@ def test_cache_normalize(self, mock_fetch):
...
+    def test_get_city_no_types(self):
+        """
+        tests the get_address function when the returned result
+        has no types field
+        """
+        city, locality = get_city(fetch_geocode_no_type("00000"))
+        self.assertEqual(city, None)
+        self.assertEqual(locality, None)
+
+    def test_get_state_no_types(self):
```
Comment (mok232) [inline external/googleapi/tests.py]: Love the tests!

**Comment by Ayushi-0903** [comment]
Looks fine to me.

File: external/googleapi/tests.py (hunk 1 above)
Comment (ab7289) [reply external/googleapi/tests.py]: 👍

**Pull request merged**